### PR TITLE
Issue #496: embedding lookup with tf.keras

### PR DESCRIPTION
**Pull request by andriy-nikolov**
Since tensorflow.keras uses the ResourceGather operation type for embedding lookup instead of GatherV2, registered gather as op handler for ResourceGather

### CODE REVIEW

**Comment by slundberg** [comment]
Thanks!! It looks good, but did you check and make sure the sum of the attribution values plus the expected value still equals the output for your example? (a good sanity check)

**Comment by andriy-nikolov** [comment]
Sorry, got distracted with other things. Yes, added a corresponding test into `test_deep.py`: a copy of `test_keras_imdb_lstm()`, but using the `tensorflow.keras` packages.

**Comment by slundberg** [comment]
thanks!

**Comment by akshayuppal3** [comment]
I am still getting the error with my model while using deep explainer...
Layer (type)                    Output Shape         Param #     Connected to                     
==================================================================================================
input_12 (InputLayer)           (None, 150, 18, 10)  0                                            
__________________________________________________________________________________________________
input_11 (InputLayer)           (None, 150, 18)      0                                            
__________________________________________________________________________________________________
reshape_23 (Reshape)            (None, 2700, 10)     0           input_12[0][0]                   
__________________________________________________________________________________________________
reshape_21 (Reshape)            (None, 2700)         0           input_11[0][0]                   
__________________________________________________________________________________________________
time_distributed_16 (TimeDistri (None, 2700, 10, 10) 1070        reshape_23[0][0]                 
__________________________________________________________________________________________________
embedding_11 (Embedding)        (None, 2700, 300)    18046800    reshape_21[0][0]                 
__________________________________________________________________________________________________
time_distributed_17 (TimeDistri (None, 2700, 60)     17040       time_distributed_16[0][0]        
__________________________________________________________________________________________________
reshape_22 (Reshape)            (None, 150, 5400)    0           embedding_11[0][0]               
__________________________________________________________________________________________________
reshape_24 (Reshape)            (None, 150, 1080)    0           time_distributed_17[0][0]        
__________________________________________________________________________________________________
masking_11 (Masking)            (None, 150, 5400)    0           reshape_22[0][0]                 
__________________________________________________________________________________________________
masking_12 (Masking)            (None, 150, 1080)    0           reshape_24[0][0]                 
__________________________________________________________________________________________________
concatenate_6 (Concatenate)     (None, 150, 6480)    0           masking_11[0][0]                 
                                                                 masking_12[0][0]                 
__________________________________________________________________________________________________
bidirectional_11 (Bidirectional (None, 150, 240)     6336960     concatenate_6[0][0]              
__________________________________________________________________________________________________
bidirectional_12 (Bidirectional (None, 150, 240)     346560      bidirectional_11[0][0]           
__________________________________________________________________________________________________
time_distributed_18 (TimeDistri (None, 150, 100)     24100       bidirectional_12[0][0]           
__________________________________________________________________________________________________
crf_6 (CRF)                     (None, 150, 2)       210         time_distributed_18[0][0]        
__________________________________________________________________________________________________
dense_layer (Dense)             (None, 150, 1)       3           crf_6[0][0]                      
__________________________________________________________________________________________________
lambda_13 (Lambda)              (None, 150, 1)       0           dense_layer[0][0]                
__________________________________________________________________________________________________
reshape_last (Reshape)          (None, 150)          0           lambda_13[0][0]                  
__________________________________________________________________________________________________
dense_3 (Dense)                 (None, 150)          22650       reshape_last[0][0]

**Comment by akshayuppal3** [comment]
I am still getting the error with my model while using deep explainer...

image
